{"metadata":{"kernelspec":{"language":"python","display_name":"Python 3","name":"python3"},"language_info":{"name":"python","version":"3.7.10","mimetype":"text/x-python","codemirror_mode":{"name":"ipython","version":3},"pygments_lexer":"ipython3","nbconvert_exporter":"python","file_extension":".py"}},"nbformat_minor":4,"nbformat":4,"cells":[{"cell_type":"markdown","source":"<br>\n<h1 style = \"font-size:60px; font-family:Garamond ; font-weight : normal; background-color: #f6f5f5 ; color : #fe346e; text-align: center; border-radius: 100px 100px;\">Introduction to single-cell RNA-seq</h1>\n<br>\n<br>\n\n<h1 style = \"font-size:40px; font-family:Garamond ; font-weight : normal; background-color: #f6f5f5 ; color : #fe346e; text-align: center; border-radius: 100px 100px;\">Bulk RNA-seq</h1>\n<br> \n\n- A major breakthrough (replaced microarrays) in the late 2000's and has been widely used since.\n- Bulk RNA-sequencing is the method of choice for <b>transcriptomics analysis</b> of pooled cell populations, tissue sections or biopsies, measuring the <b>average expression level</b> for each gene across a large population of input cells.\n- However they do not differentiate among cell types within the sample, rather they give a view of gene expression within a whole organ or tissue type.\n- Useful for <b>comparative transcriptomics</b>, e.g. quantifying expression signatures in samples of the same tissue from different species.\n- However, it is insufficient in providing insights into the stochastic nature of gene expression.","metadata":{"_uuid":"8f2839f25d086af736a60e9eeb907d3b93b6e0e5","_cell_guid":"b1076dfc-b9ad-4769-8c92-a6c4dae69d19"}},{"cell_type":"markdown","source":"<h1 style = \"font-size:40px; font-family:Garamond ; font-weight : normal; background-color: #f6f5f5 ; color : #fe346e; text-align: center; border-radius: 100px 100px;\">scRNA-seq </h1>\n<br> \n\n- A <b>new technology</b>, first publication by Tang et al. in 2009\n- Measures the <b>distribution of expression levels</b> for each gene across a population of cells\n- Allows us to study new biological questions in which <b>cell-specific changes in transcriptome are important</b>, e.g. cell type identification, heterogeneity of cell responses, stochasticity of gene expression, inference of gene regulatory networks across the cells.\n- Several computational analysis methods from bulk RNA-seq can be used.\n- In most cases computational analysis requires <b>adaptation</b> of the existing methods or <b>development</b> of new ones","metadata":{}},{"cell_type":"markdown","source":"- Single cell sequencing examines the sequence information from individual cells with optimized next-generation sequencing (NGS) technologies, providing a higher resolution of cellular differences and a better understanding of the function of an individual cell in the context of its microenvironment. \n- For example, in cancer, sequencing the DNA of individual cells can give information about mutations carried by small populations of cells. \n- In development, sequencing the RNAs expressed by individual cells can give insight into the existence and behavior of different cell types. \n- In microbial systems, a population of the same species can appear to be genetically clonal, but single-cell sequencing of RNA or epigenetic modifications can reveal cell-to-cell variability that may help populations rapidly adapt to survive in changing environments.","metadata":{}},{"cell_type":"code","source":"import PIL\nfrom PIL import Image\nImage.open(\"../input/scrnaseq-analysis-model-outputs/RNA-Seq_workflow-5.pdf.jpg\")","metadata":{"execution":{"iopub.status.busy":"2021-09-20T05:35:37.327511Z","iopub.execute_input":"2021-09-20T05:35:37.327828Z","iopub.status.idle":"2021-09-20T05:35:37.407101Z","shell.execute_reply.started":"2021-09-20T05:35:37.327796Z","shell.execute_reply":"2021-09-20T05:35:37.40649Z"},"trusted":true},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":"Overall, experimental scRNA-seq protocols are similar to the methods used for bulk RNA-seq.","metadata":{}},{"cell_type":"markdown","source":"## Computational Analysis \nThis notebook is concerned with the computational analysis of the data obtained from scRNA-seq experiments. \n- The first steps (yellow) are general for any highthroughput sequencing data. \n- Later steps (orange) require a mix of existing RNASeq analysis methods and novel methods to address the technical difference of scRNASeq. \n- Finally the biological interpretation (blue) should be analyzed with methods specifically developed for scRNASeq.","metadata":{}},{"cell_type":"code","source":"Image.open(\"../input/scrnaseq-analysis-model-outputs/Screenshot 2021-09-12 142058.jpg\")","metadata":{"execution":{"iopub.status.busy":"2021-09-20T05:36:00.141603Z","iopub.execute_input":"2021-09-20T05:36:00.142288Z","iopub.status.idle":"2021-09-20T05:36:00.185989Z","shell.execute_reply.started":"2021-09-20T05:36:00.142246Z","shell.execute_reply":"2021-09-20T05:36:00.185047Z"},"trusted":true},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"Image.open(\"../input/scrnaseq-analysis-model-outputs/msb188746-fig-0001-m.jpg\")","metadata":{"execution":{"iopub.status.busy":"2021-09-20T05:36:18.818854Z","iopub.execute_input":"2021-09-20T05:36:18.821203Z","iopub.status.idle":"2021-09-20T05:36:19.52126Z","shell.execute_reply.started":"2021-09-20T05:36:18.821157Z","shell.execute_reply":"2021-09-20T05:36:19.520243Z"},"trusted":true},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":"## Challenges \nThe main difference between bulk and single cell RNA-seq is that each sequencing library represents a single cell, instead of a population of cells. Therefore, significant attention has to be paid to comparison of the results from different cells (sequencing libraries). The main sources of discrepancy between the libraries are:\n\n- Amplification (up to 1 million fold)\n- Gene 'dropouts' in which a gene is observed at a moderate expression level in one cell but is not detected in another cell (Kharchenko, 2014).\n","metadata":{}},{"cell_type":"markdown","source":"https://chanzuckerberg.github.io/scRNA-python-workshop/preprocessing/01-basic-qc.html","metadata":{}}]}